youtube_dl/extractor/vlive.py

```diff
@@ -31,16 +34,73 @@ def _real_extract(self, url):
 
-        long_video_id = self._search_regex(
-            r'vlive\.tv\.video\.ajax\.request\.handler\.init\(\s*"[0-9]+"\s*,\s*"[^"]*"\s*,\s*"([^"]+)"',
-            webpage, 'long video id')
+        status = self._download_json(
+            'http://www.vlive.tv/video/status?videoSeq=%s' % video_id,
+            video_id, 'Downloading JSON status',
+            headers={'Referer': url}).get('status')
 
-        key = self._search_regex(
-            r'vlive\.tv\.video\.ajax\.request\.handler\.init\(\s*"[0-9]+"\s*,\s*"[^"]*"\s*,\s*"[^"]+"\s*,\s*"([^"]+)"',
-            webpage, 'key')
+        video_params = self._search_regex(
+            r'vlive\.tv\.video\.ajax\.request\.handler\.init\((.*)\)',
+            webpage, 'video params')
+        live, long_video_id, key = re.split(r'"\s*,\s*"', video_params)[1:4]
 
+        if status == 'LIVE_ON_AIR' or status == 'BIG_EVENT_ON_AIR':
+            live_params = json.loads(json.loads('"%s"' % live))
+            return self._live(video_id, webpage, live_params)
+        elif status == 'VOD_ON_AIR' or status == 'BIG_EVENT_INTRO':
+            return self._replay(video_id, webpage, long_video_id, key)
+        elif status == 'LIVE_END':
+            raise ExtractorError('Uploading for replay. Please wait…',
+                                 expected=True)
+        elif status == 'COMING_SOON':
+            raise ExtractorError('Coming soon!', expected=True)
+        else:
+            raise ExtractorError('Unknown status %s' % status)
+
+    def _get_common_fields(self, webpage):
         title = self._og_search_title(webpage)
+        creator = self._html_search_regex(
+            r'<div[^>]+class="info_area"[^>]*>\s*<a\s+[^>]*>([^<]+)',
+            webpage, 'creator', fatal=False)
+        thumbnail = self._og_search_thumbnail(webpage)
+        return {
+            'title': title,
+            'creator': creator,
+            'thumbnail': thumbnail,
+        }
+
+    def _get_sub_url(self, video_url, lang):
+        url = video_url.split('/playlist.m3u8')[0]
+        url += '/subtitlelist_l%s.m3u8' % lang
+        return url
+
+    def _live(self, video_id, webpage, live_params):
+        formats = [{
+            'url': vid['cdnUrl'],
+            'ext': 'mp4',
+            'format_id': vid.get('name'),
+            'quality': int_or_none(vid.get('name')),
+            'width': int_or_none(vid.get('width')),
+            'height': int_or_none(vid.get('height')),
+        } for vid in live_params.get('resolutions', [])]
+        self._sort_formats(formats)
+
+        subtitles = {
+            'eng': [{'ext': 'vtt', 'url': self._get_sub_url(f['url'], 'eng')}
+                    for f in formats],
+            'zcn': [{'ext': 'vtt', 'url': self._get_sub_url(f['url'], 'zcn')}
+                    for f in formats],
+            'vie': [{'ext': 'vtt', 'url': self._get_sub_url(f['url'], 'vie')}
+                    for f in formats],
+        }
```

Comment:
I haven't found anything similar in Live API, seems like the only way to determine subtitle status is to download `playlist.m3u8` and parse it manually. It looks like this:

```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="subs",NAME="Vietnamese",FORCED=NO,AUTOSELECT=YES,URI="subtitlelist_lvie.m3u8",LANGUAGE="vie"
#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="subs",NAME="English",FORCED=NO,AUTOSELECT=YES,URI="subtitlelist_leng.m3u8",LANGUAGE="eng"
#EXT-X-MEDIA:TYPE=SUBTITLES,GROUP-ID="subs",NAME="zcn",FORCED=NO,AUTOSELECT=YES,URI="subtitlelist_lzcn.m3u8",LANGUAGE="zcn"
#EXT-X-STREAM-INF:BANDWIDTH=1464974,CODECS="avc1.66.22,mp4a.40.2",RESOLUTION=640x368,SUBTITLES="subs"
chunklist.m3u8?__agda__=1461853927_d2d91835e251112f931019235adf7e99
```

Are you ok with that?